### PR TITLE
Refactor ListProject calls to support the v4 API

### PR DESCRIPTION
**Pull request by CHERTS**
Hi

I fixed GitLab APIv4 support

### CODE REVIEW

**Comment by svanharmelen** [comment]
@CHERTS besides that the Travis checks are failing, this PR does not provide a proper fix.

Have a look at this [page](https://docs.gitlab.com/ce/api/v3_to_v4.html) to see what changed. If you look at the changes related to the endpoints you deleted, you'll notice they should not be deleted, but adjusted:

https://docs.gitlab.com/ce/api/v3_to_v4.html
image

We keep track of the V4 support using issue #151 which contains the same list, but with tickboxes to tick off items that are done.

Are you willing to update this PR so it properly fixes the changes? If not I'll close this PR. Thx!

**Comment by CHERTS** [comment]
I fixed the projects_test, please run a check in Travis

**Comment by svanharmelen** [comment]
@CHERTS please read my comment again... The failing tests don't matter much since the changes themselves just don't make sense. You're not fixing anything, you're just deleting things.

The endpoints that you deleted still exist in the V4 API, but they are used slightly different. So the code should be updated/changed, not deleted.

**Comment by CHERTS** [comment]
In APIv4, there are no endpoints "/project/owned" "/project/all" "/projects/starred" and this can easily be seen in this query:

curl --request GET -H "PRIVATE-TOKEN: XXXXX" -H "Accept: application/json" -H "Content-Type:application/json" https://gitlab.xxxxx.ru/api/v4/projects/all | jq .

{
  "message": "404 Project Not Found"
}

This endpoints only exist in APIv3

For example, APIv3 had:
curl --request GET -H "PRIVATE-TOKEN: XXXXX" -H "Accept: application/json" -H "Content-Type:application/json" https://gitlab.xxxxx.ru/api/v4/projects/owned | jq .

In APIv4 was:
curl --request GET -H "PRIVATE-TOKEN:  XXXXX" -H "Accept: application/json" -H "Content-Type:application/json" https://gitlab.xxxxx.ru/api/v4/projects?owned=true | jq .

I fix in the code "type ListProjectsOptions struct" and removed the func ListOwnedProjects, ListStarredProjects, ListAllProjects. These functions are no longer needed for APIv4

**Comment by CHERTS** [comment]
If we do not understand each other, then please close the my pull request.

**Comment by svanharmelen** [comment]
@CHERTS I understand, but I don't agree that we should just remove these functions...

The functions should remain in the code, but the function bodies should be changed so they no longer call the old endpoints, but use the new format (e.g. `/projects?owned=true`).

As the endpoint is maybe gone, but the functionality isn't. You can still list all owned or starred projects, but the API just expects a different call for that...

Do you see my point?

**Comment by CHERTS** [comment]
I understand that deleting these functions (ListOwnedProjects, ListStarredProjects, ListAllProjects) will violate the compatibility, but this is normal if the API changes a lot, which is the case.

Why duplicate the functions of ListOwnedProjects, ListStarredProjects, ListAllProjects if the use of the function ListProjects does the same?

File: gitlab.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ import (
 )
 
 const (
-	libraryVersion = "0.1.1"
+	libraryVersion = "0.1.2"
```
Comment (svanharmelen) [inline gitlab.go]: Please make this `0.2.0`

File: gitlab.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
...
 	defaultBaseURL = "https://gitlab.com/api/v3/"
```
Comment (svanharmelen) [inline gitlab.go]: I noticed you first changed this `v4`, but changed it back again. I guess it's fine to change this to `v4` as this branch is for the v4 API.

File: projects.go [hunk 3]
```diff
@@ -168,7 +168,9 @@ type ListProjectsOptions struct {
 	Search     *string               `url:"search,omitempty" json:"search,omitempty"`
 	Simple     *bool                 `url:"simple,omitempty" json:"simple,omitempty"`
 	Visibility *VisibilityLevelValue `url:"visibility,omitempty" json:"visibility,omitempty"`
-	Statistics *bool                 `url:"statistics,omitempty" json:"statistics,omitempty"`
```
Comment (svanharmelen) [inline projects.go]: Looking at the docs, I notice that `Statistics` is still a valid option. So could you please add back `Statistics to the bottom of this list? And to satisfy my OCD, could you also relocate `Visibility` to be the last entry? Thx!

```
...
Owned
Membership
Starred
Statistics
Visibility
```

File: projects_test.go [hunk 4]
```diff
@@ -22,12 +22,14 @@ func TestListProjects(t *testing.T) {
 			"search":     "query",
 			"simple":     "true",
 			"visibility": "public",
-			"statistics": "true",
+			"owned": "false",
+			"membership": "false",
+			"starred": "false",
```
Comment (svanharmelen) [inline projects_test.go]: Please update the tests as well so they include `statistics` again...

**Comment by CHERTS** [comment]
Ok, fixed libraryVersion and back Statistics

**Comment by svanharmelen** [comment]
Thanks @CHERTS!

**Pull request merged**